### PR TITLE
Core validation: add a warning if some VkRenderPassBeginInfo.pClearValues are ignored

### PR DESCRIPTION
**Pull request by scygan**
Vulkan spec does permit clearValueCount to be greater than number of values used
by clear ops. However we have seen apps that pass garbage as clearValueCount - the
best we can do in this situation is to emit a warning.

### CODE REVIEW

File: tests/layer_validation_tests.cpp [hunk 1]
```diff
@@ -7868,6 +7868,65 @@ TEST_F(VkLayerTest, RenderPassClearOpMismatch) {
...
+    rpci.pAttachments = &attach_desc;
+    rpci.sType = VK_STRUCTURE_TYPE_RENDER_PASS_CREATE_INFO;
+    VkRenderPass rp;
+    vkCreateRenderPass(m_device->device(), &rpci, NULL, &rp);
+
+    VkCommandBufferInheritanceInfo hinfo = {};
+    hinfo.sType = VK_STRUCTURE_TYPE_COMMAND_BUFFER_INHERITANCE_INFO;
+    hinfo.renderPass = VK_NULL_HANDLE;
+    hinfo.subpass = 0;
+    hinfo.framebuffer = VK_NULL_HANDLE;
```
Comment (chrisforbes) [inline tests/layer_validation_tests.cpp]: You don't need this structure at all -- this is a primary command buffer; VkCommandBufferBeginInfo::pInheritanceInfo is ignored.

File: layers/core_validation.cpp [hunk 2]
```diff
@@ -10705,6 +10705,17 @@ CmdBeginRenderPass(VkCommandBuffer commandBuffer, const VkRenderPassBeginInfo *p
                             pRenderPassBegin->clearValueCount, clear_op_size, reinterpret_cast<uint64_t &>(renderPass),
                             clear_op_size, clear_op_size - 1, validation_error_map[VALIDATION_ERROR_00442]);
             }
+            if (clear_op_size < pRenderPassBegin->clearValueCount) {
+                skip_call |=
+                    log_msg(dev_data->report_data, VK_DEBUG_REPORT_WARNING_BIT_EXT, VK_DEBUG_REPORT_OBJECT_TYPE_RENDER_PASS_EXT,
+                            reinterpret_cast<uint64_t &>(renderPass), __LINE__, DRAWSTATE_RENDERPASS_TOO_MANY_CLEAR_VALUES, "DS",
+                            "In vkCmdBeginRenderPass() the VkRenderPassBeginInfo struct has a clearValueCount of %u but only "
+                            "first %u "
```
Comment (chrisforbes) [inline layers/core_validation.cpp]: Reflow this?

**Comment by mark-lunarg** [comment]
Removed the unnecessary structure in the test, forced CF to reflow the error messages, and fixed an unexpected error in the test.  Running through the farm now.

**Comment by mark-lunarg** [comment]
Also fixed up the layer details doc.  Pushed to Master.  Thanks Slawomir.

**Pull request closed without merging**